### DISCUSSION TITLE
Automate/On demand Flow Trigger

### DISCUSSION THREAD

**Discussion by handsome-nium** [postmanlabs/postman-flows#241] [Ideas]
It would be nice if you could trigger the flow in an automated way, or just via Postman API. It will allow you to:

- create a cron job for triggering a flow, and be notified if something went wrong. 
- Request Postman API for triggering a flow directly from a CI/CD, and get the results.

This will be really useful, specially if you are using documentation with collection. This will allow creating integration tests, focused on documentation-collection, to ensure that your documentation is always up to date with your code.

**Comment by samay-v** [comment]
Hey @handsome-nium 
This is definitely a functionality we want to introduce for flows. It'll be very useful if you can also explain one of your usecase in detail. 
There is a previous discussion around this here as well https://github.com/postmanlabs/postman-flows/discussions/19#discussioncomment-1458902

**Comment by handsome-nium** [comment]
+to Adri, Jamie

Hi Samay,

I would like to introduce you to Adri and Jamie, the two engineers working directly with Postman. I think they can provide the detail you seek.

@Adrian Nunez ***@***.******@***.*** Kichuk ***@***.***> Please provide insight into how we want to use flow to validate documentation and our concerns about having two integration tests i.e., one that runs in postman and one that runs in our CI/CD.

Thanks.

g


From: Samay Verma ***@***.***>
Date: Monday, February 7, 2022 at 12:39 AM
To: postmanlabs/postman-flows ***@***.***>
Cc: Geoff Greenberg (NIUM) ***@***.***>, Mention ***@***.***>
Subject: Re: [postmanlabs/postman-flows] Automate/On demand Flow Trigger (Discussion #241)
CAUTION: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.


Hey @handsome-nium<https://github.com/handsome-nium>
This is definitely a functionality we want to introduce for flows. It'll be very useful if you can also explain one of your usecase in detail.
There is a previous discussion around this here as well #19 (comment)<https://github.com/postmanlabs/postman-flows/discussions/19#discussioncomment-1458902>

—
Reply to this email directly, view it on GitHub<https://github.com/postmanlabs/postman-flows/discussions/241#discussioncomment-2123559>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWODIBYHRBURBAZZON3NW7DUZ6ALXANCNFSM5NSDBC3Q>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by handsome-nium** [reply]
Hi @samay-v , thank you for replying the Idea discussion!

About using Postman flows, we are trying to achieve two things:

- Validate that our API documentation is up to date and is correct. i.e. functional with our code. As we can use directly collections on flows, they will always make requests to our environment using the most recent documentation collection changes. 
- Create integration tests to validate our business logic flows. 

Our main concern, is having the ability of running this directly through our CI/CD. With a direct integration Postman-Bitbucket, or triggering flows on demand via Postman API. With this, we could use postman flows to cover both  previous exposed things.

I saw in the thread #19 that this is on your Roadmap which is awesome! Do you have any update on this or an estimated ETA?

Thanks! Regards,
Adri.
cc. @handsome-nium

**Reply by saswatds** [reply]
Hey Adri, Thanks for your valuable feedback. This will surely help us make more informed decisions. We are currently working on integrating flows with rest of the postman eco-system to become generally available. Keep an eye out on #19 . I will post an update on that thread when we make any progress.

**Reply by Leighdarlow** [reply]
Hi, we'd also find this very useful, is there an ETA, thanks, Leigh
